shared/src/commonMain/kotlin/org/hyperskill/app/next_learning_activity_widget/presentation/NextLearningActivityWidgetFeature.kt

```diff
@@ -0,0 +1,79 @@
+package org.hyperskill.app.next_learning_activity_widget.presentation
+
+import org.hyperskill.app.analytic.domain.model.AnalyticEvent
+import org.hyperskill.app.learning_activities.domain.model.LearningActivity
+import org.hyperskill.app.study_plan.domain.model.StudyPlan
+import org.hyperskill.app.study_plan.widget.presentation.model.LearningActivityTargetViewAction
+
+object NextLearningActivityWidgetFeature {
+    data class State(
+        internal val studyPlan: StudyPlan?,
+        val contentState: ContentState
+    ) {
+        internal val isRefreshing: Boolean
+            get() = contentState is ContentState.Content && contentState.isRefreshing
+    }
+
+    internal fun initialState(): State =
+        State(studyPlan = null, contentState = ContentState.Idle)
+
+    sealed interface ContentState {
+        object Idle : ContentState
+        object Loading : ContentState
+        object NetworkError : ContentState
+        object Empty : ContentState
+        data class Content(
+            val learningActivity: LearningActivity,
+            val isRefreshing: Boolean = false
+        ) : ContentState
+    }
+
+    sealed interface ViewState {
+        object Idle : ViewState
+        object Loading : ViewState
+        object NetworkError : ViewState
+        object Empty : ViewState
+        data class Content(
+            val title: String,
+            val subtitle: String?,
+            val isIdeRequired: Boolean,
+            val progress: Int,
+            val formattedProgress: String?
+        ) : ViewState
+    }
+
+    sealed interface Message {
+        object RetryContentLoading : Message
+        object NextLearningActivityClicked : Message
+    }
+
+    internal sealed interface InternalMessage : Message {
+        data class Initialize(val forceUpdate: Boolean = false) : InternalMessage
```

Comment:
This is HomeFeature legacy, home can initialize (force reload data) itself and subfeatures:
1. For the first time - initialize
2. When the app enters the foreground from the background - it forces the home to reload
3. When a problem of the day card requests refresh
4. When reloading home data after error

So, it's much simpler to use initialize message with a forceUpdate parameter like other subfeatures, instead of trying to refactor all of this.